### ISSUE TITLE
problem installing ioos environment on Ubuntu

### ISSUE THREAD

**Issue by emiliom** [ioos/conda-recipes#251]
I have a few minutes of playing around and decided to try to follow @ocefpaf's instructions to install the complete IOOS environment. I followed the [instructions from his presentation](https://ocefpaf.github.io/DMAC_presentation/binstar/slides.html#/how-to-use-it). BTW, in the statement
`wget http://bit.ly/ioos_req` the file that's downloaded is called `ioos_req`, so I renamed it to `ioos_req.txt` to work with the create statement `conda create --yes -n IOOS --file ioos_req.txt python=2.7`.

`conda create` gave me an error: 
```
Error: No packages found matching: cartopy >=0.11
```
FYI, I originally installed conda from Anaconda 1.9.1. My current version of conda (`conda --version`) is 3.0.6.

I probably won't be able to respond quickly this week, but I figured it'd be helpful to post this now, and better to do it via github rather than emails. Thanks, and great seeing @rsignell-usgs and meeting Filipe and hearing about all this this week!

**Comment by ocefpaf** [comment]
Thanks for testing!

> I have a few minutes of playing around and decided to try to follow
@ocefpaf's instructions to install the complete IOOS environment. I
followed the instructions from his presentation. BTW, in the statement
> wget http://bit.ly/ioos_req the file that's downloaded is called
ioos_req, so I renamed it to ioos_req.txt to work with the create statement
conda create --yes -n IOOS --file ioos_req.txt python=2.7.

Ah... That's an issue when using URL shorteners. I will fix the
instructions to redirect to the desired filename.

> conda create gave me an error:
>
> Error: No packages found matching: cartopy >=0.11

Can you send me the result of "conda info"?

> FYI, I originally installed conda from Anaconda 1.9.1. My current version
of conda (conda --version) is 3.0.6.

You should be OK with that.

> I probably won't be able to respond quickly this week, but I figured it'd
be helpful to post this now, and better to do it via github rather than
emails. Thanks, and great seeing @rsignell-usgs and meeting Filipe and
hearing about all this this week!

It was great to meet you too.

**Comment by emiliom** [comment]
Here are the results of "conda info":
```
Current conda install:

             platform : linux-64
        conda version : 3.0.6
       python version : 2.7.6.final.0
     root environment : /usr/anaconda/anaconda  (writable)
  default environment : /usr/anaconda/anaconda
     envs directories : /usr/anaconda/anaconda/envs
        package cache : /usr/anaconda/anaconda/pkgs
         channel URLs : https://conda.binstar.org/ioos/linux-64/
                        http://repo.continuum.io/pkgs/free/linux-64/
                        http://repo.continuum.io/pkgs/pro/linux-64/
          config file : /home/mayorga/.condarc
    is foreign system : False
```

Thanks!

**Comment by ocefpaf** [comment]
Humn... Not sure what is happening.  Can you send me the output of `conda search cartopy`?  

Here  is what we expect:

```shell
Fetching package metadata: ......
cartopy                      0.12.dev             np19py27_0  ioos            
                             0.12.dev             np18py27_0  ioos            
                             0.12.0rc1            np19py27_0  ioos            
                             0.12.0               np19py27_0  ioos            
                             0.12.0               np19py27_1  ioos            
                          .  0.12.0               np19py27_2  ioos 
```

**Comment by emiliom** [comment]
Here is `conda search cartopy` output that looks identical to what you listed, followed by the `conda create` statement that leads to the same error I reported.
```
mayorga@emilioll:/usr/anaconda/anaconda/envs$ conda search cartopy
cartopy                      0.12.dev             np18py27_0  ioos            
                             0.12.dev             np19py27_0  ioos            
                             0.12.0rc1            np19py27_0  ioos            
                             0.12.0               np19py27_0  ioos            
                             0.12.0               np19py27_1  ioos            
                             0.12.0               np19py27_2  ioos       

mayorga@emilioll:/usr/anaconda/anaconda/envs$ conda create --yes -n IOOS --file ioos_req.txt python=2.7
Error: No packages found matching: cartopy >=0.11
```

**Comment by ocefpaf** [comment]
As you can see conda is not helpful :unamused: Cartopy is there, but somehow it cannot install it on your machine.

Can you try: `conda install cartopy` with a combination of `--debug` and/or `--verbose` options so we can try to get to the bottom of this.

Also, you might want to try `conda update conda`, maybe that is just a bug that was already fixed.

**Comment by emiliom** [comment]
I looked into `conda update conda`, but decided not to do it b/c I don't know what impact all those updates would have (if any) on my existing environments.

`conda install` doesn't have --debug or --verbose arguments. I tried to install cartopy as follows, and ran into the new error you see at the bottom:
```
$ conda create --yes -n IOOScartopy1 python=2.7
$ source activate IOOScartopy1
$ conda install -n IOOScartopy1 cartopy
Error: No packages found matching: fontconfig  >=2.11.1
```
Thanks.

**Comment by ocefpaf** [comment]
> I looked into conda update conda, but decided not to do it b/c I don't know what impact all those updates would have (if any) on my existing environments.

It *should* not affect your current envs, unless you update them later...

A quick search shows `fontconfig` in the default channel for me, and I guess it might show it there for you too. (Try `conda search fontconfig` to confirm that.) 

My guess is that we are hitting a bug in your conda version...  Not sure where to go from here.

**Comment by emiliom** [comment]
Ok, I went ahead and did `conda update conda`. 
```
$ source activate IOOScartopy1
$ conda install -n IOOScartopy1 cartopy
Fetching package metadata: ......
Solving package specifications: .
Error: could not import pycosat (required for dependency resolving)
```
But pycosat 0.6.1 **was** installed during `conda update conda`. I also tried the following, and ran into the same error:
```
$ conda create --yes -n IOOS --file ioos_req.txt python=2.7
Fetching package metadata: ......
Solving package specifications: .
Error: could not import pycosat (required for dependency resolving)
```
Sigh. This is getting disappointing :(

**Comment by ocefpaf** [comment]
Indeed!  (Now you know why I was against conda at first :wink:)

But now I think it is time to take this thread to the conda mailing list.  I can do that for you.

**Comment by ocefpaf** [comment]
BTW: pycosat should be the solution to the versioning issue we are seeying here, that is why the latest conda depends on it.

**Comment by emiliom** [comment]
Thanks for offering to take it to the conda mailing list. I would stop my own attempts there if that's what was needed ...

**Comment by abirger** [comment]
@ocefpaf , kind of off-topic, but the instruction for Windows works pretty well, and does not generate any problem at all. I did it with Anaconda 2.1, and 2.2, and both times it went smoothly.

**Comment by rsignell-usgs** [comment]
@abirger , thanks for that report.   @emiliom, I guess you should have bought a PC!  ;-)
The linux side is working well also, so just problems with Mac.

**Comment by ocefpaf** [comment]
@rsignell-usgs @emiliom problems are with Linux and not Mac.

I am writing to the mailing list now to report what happened and ask for some guidance.  The issue comes from the fact that he is updating an old conda and the version solver is buggy.

**Comment by rsignell-usgs** [comment]
I like that you can delete comments from Github.  :smile_cat:

**Comment by ocefpaf** [comment]
@rsignell-usgs that is a :facepunch: right in the :nose: of provenance.

And this is me  :imp: looking at you!

**Comment by emiliom** [comment]
I'm happy to report that I was able to install and test the IOOS env on my Ubuntu laptop! After a couple of email exchanges with @ocefpaf, I decided to leave my current Anaconda/conda installation alone, and install a new, fresh conda separately using miniconda. I don't use the Anaconda stuff as my default, so that made things easier. I followed the [Linux64 instructions page](https://github.com/ioos/conda-recipes/wiki/Setting-up-the-IOOS-Python-environment-for-Linux64), and everything was largely seamless. Then I was able to successfully run both of the test notebooks from the [Testing page](https://github.com/ioos/conda-recipes/wiki/Testing-the-IOOS-environment). Major coolness! 

A minor comment, about the Anaconda launcher. I had never used it. Nifty! It didn't get installed by default by miniconda, so I installed it per the instructions in Testing. A launcher icon wasn't created. The only thing that didn't work as stated in the instructions was starting the launcher at the terminal; typing `launcher` didn't work. I looked around, and found that launcher was located in the bin directory of my IOOS env, so this worked at the terminal: `$HOME/miniconda/envs/IOOS/bin/launcher`.

Thanks for your fantastic work, @ocefpaf and @rsignell-usgs! Now I'm ready to partee.
